internal/servers/controller/handlers/accounts/account_service.go

```diff
@@ -202,26 +240,43 @@ func (s Service) SetPassword(ctx context.Context, req *pbs.SetPasswordRequest) (
 		return nil, err
 	}
 	u.Scope = authResults.Scope
-	u.AuthorizedActions = authResults.FetchActionSetForId(ctx, u.Id, IdActions).Strings()
+	u.AuthorizedActions = authResults.FetchActionSetForId(ctx, u.Id, IdActions[auth.SubtypeFromId(u.Id)]).Strings()
 	return &pbs.SetPasswordResponse{Item: u}, nil
 }
 
 func (s Service) getFromRepo(ctx context.Context, id string) (*pb.Account, error) {
```

Comment:
mind adding tests for IsNotFound for both subtypes?

Reply:
Added